app.js

```diff
@@ -1,83 +1,77 @@
+/*eslint no-console: ["error", { allow: ["warn", "error"] }] */
+/*eslint-env node*/
+
 const express = require('express');
 const app = express();
-const morgan = require('morgan');
 const bodyParser = require('body-parser');
+const alasql = require('alasql');
+
 
 app.use(express.static('public'));
 
-const PORT = process.env.PORT || 4001;
-
-const items = [
-  {
-    id: 1,
-    name: 'Bread',
-    quantity: '1',
-    unit: 'Kg'
-  },
-  {
-    id: 2,
-    name: 'Milk',
-    quantity: '1',
-    unit: 'L'
-  },
-  {
-    id: 3,
-    name: 'Pasta',
-    quantity: '2',
-    unit: 'Kg'
-  }
-];
-let nextId = 4;
+const PORT = process.env.PORT || 4002;
+
+alasql('CREATE TABLE items (id int, name varchar(255), quantity int, unit varchar(25))');
+
+alasql('INSERT INTO items VALUES (\'1\',\'Bread\',\'1\',\'Kg\')');
+alasql('INSERT INTO items VALUES (\'2\',\'Rice\',\'2\',\'Kg\')');
+alasql('INSERT INTO items VALUES (\'3\',\'Pudding\',\'2\',\'Kg\')');
 
 // Body-parsing Middleware
 app.use(bodyParser.json());
 
-// Logging Middleware
-if (!process.env.IS_TEST_ENV) {
-  app.use(morgan('dev'));
-}
-
 // Get all Items
 app.get('/items/', (req, res, next) => {
-  res.send(items);
+	const items = alasql('SELECT * FROM items');
+	res.send(items);
+	next();
 });
 
 // Get a single Item
 app.get('/items/:itemId', (req, res, next) => {
-  const itemId = Number(req.params.itemId);
-  const itemIndex = items.findIndex(item => item.id === itemId);
-  if (itemIndex === -1) {
-  res.status(201).send(itemIndex);
-  }
-  res.send(items[itemIndex]);
+	const itemId = Number(req.params.itemId);
+	const item = alasql('SELECT * FROM items WHERE id='+ itemId);
+	const itemIndex = item.findIndex(item => item.id === itemId);
+	if (itemIndex === -1) {
+		return res.status(404).send('item not found');
+	}
+	res.send(item);
+	next();
 });
 
+
 // Create a new item
 app.post('/items/', (req, res, next) => {
-  const newitem = req.body;
-  newitem.id = nextId++;
-  items.push(newitem);
-  res.status(201).send(newitem);
+	const newitem = req.body;
+	alasql('SELECT * INTO items FROM ?',[[newitem]]);
+	res.status(201).send(newitem);
+	next();
 });
 
-// Delete a Item
+// Delete am Item
 app.delete('/items/:itemId', (req, res, next) => {
-  const itemId = Number(req.params.itemId);
-  const itemIndex = items.findIndex(item => item.id === itemId);
-  if (itemIndex === -1) {
-    return res.status(404).send('item not found');
-  }
-  items.splice(itemIndex, 1);
-  res.status(204).send();
+	const itemId = Number(req.params.itemId);
+	const item = alasql('SELECT * FROM items WHERE id='+ itemId);
+	alasql('DELETE FROM items WHERE id='+ itemId);
+	const itemIndex = item.findIndex(item => item.id === itemId);
+	if (itemIndex === -1) {
+		return res.status(404).send('item not found');
+	}
+
+	res.status(204).send();
+	next();
 });
 
 // Delete all Items
 app.delete('/items/', (req, res, next) => {
-  items.splice(0, items.length);
-  res.status(204).send();
+	alasql('DELETE FROM items');
```

Comment:
does it actually work?

Reply:
Yes, it does. I have tested it manualy in Restlet Client and and wrote test for it, whitch is passing.